dds/idl/dds_generator.h

```diff
@@ -612,7 +613,7 @@ void generateCaseBody(
     const char* breakString = generateBreaks ? "    break;\n" : "";
     std::string intro;
     if (commonFn2) {
-      const unsigned id = be_global->get_id(NULL, branch, 0);
+      const unsigned id = be_global->get_id(NULL, branch, default_id);
```

Comment:
You will need to pass in the union. It's not used at the moment, but will be needed in the future.